codechecker_common/report.py

```diff
@@ -25,7 +25,7 @@ class Report(object):
     from the path section for easier skip/suppression handling
     and result processing.
     """
-    def __init__(self, main, bugpath, files):
+    def __init__(self, main, bugpath, files, metadata):
```

Comment:
`metadata` could have `None` as default value. It is called with `None` several times.

Reply:
I set this parameter default value to None.

By the way we call this with None in two places but in the following way:
```
metadata = None
Report(..., metadata)
```
So from this I can see explicitly that no metadata is given to the Report class.

Reply:
I see. This two-line version has documentation reasons instead of `Report(..., None)`. You may consider using `Report(..., metadata=None)` if you want to see parameter name at function call.